dbms/src/DataStreams/WindowTransformAction.cpp

```diff
@@ -660,7 +660,7 @@ std::tuple<RowNumber, bool> WindowTransformAction::stepToStartForRangeFrame()
 
 std::tuple<RowNumber, bool> WindowTransformAction::stepToEndForRangeFrame()
 {
-    if (!window_description.frame.end_preceding && !partition_ended)
+    if (!partition_ended)
```

Comment:
Better add comments:
```
Still return false even when end_preceding is true, because there may still same rows as current_row in the next block.
So we have to continue until got partition end(or got the first different row with current_row).
```

Reply:
> Better add comments: `Still return false even when end_preceding is true, because there may still same rows as current_row in the next block. So we cannot stop find the frame end until got the partition end(or got the first row that is greater than current_row).`

added